### PR TITLE
OS: add support for Proxmox VE

### PR DESCRIPTION
**Pull request by ThomasLamprecht**
## Description

Only fill in the fields below if relevant.

### CODE REVIEW

**Comment by dylanaraps** [comment]
Thanks for the PR!

- Lines must be kept below `100` chars long.
- Is the extra detection needed?
    - Does Neofetch not detect Proxmox correctly without the extra detection stuff?
- Remove the 1 space gap between the art and the variables.
- You have too many color variables, the art can be simplified.
    - Color formatting follows line breaks.

Simplified art:

```sh

${c1}         .://:`              `://:.
       `hMMMMMMd/          /dMMMMMMh`
        `sMMMMMMMd:      :mMMMMMMMs`
`-/+oo+/:`.yMMMMMMMh-  -hMMMMMMMy.`${c2}:/+oo+/-`
`:oooooooo/${c1}`-hMMMMMMMyyMMMMMMMh-`${c2}/oooooooo:`
  `/oooooooo:${c1}`:mMMMMMMMMMMMMm:`${c2}:oooooooo/`
    ./ooooooo+-${c1} +NMMMMMMMMN+ ${c2}-+ooooooo/.
      .+ooooooo+-${c1}`oNMMMMNo`${c2}-+ooooooo+.
        -+ooooooo/.${c1}`sMMs`${c2}./ooooooo+-
          :oooooooo/${c1}`..`${c2}/oooooooo:
          :oooooooo/`${c1}..${c2}`/oooooooo:
        -+ooooooo/.`${c1}sMMs${c2}`./ooooooo+-
      .+ooooooo+-`${c1}oNMMMMNo${c2}`-+ooooooo+.
    ./ooooooo+-${c1} +NMMMMMMMMN+ ${c2}-+ooooooo/.
  `/oooooooo:`${c1}:mMMMMMMMMMMMMm:${c2}`:oooooooo/`
`:oooooooo/`${c1}-hMMMMMMMyyMMMMMMMh-${c2}`/oooooooo:`
`-/+oo+/:`${c1}.yMMMMMMMh-  -hMMMMMMMy.${c2}`:/+oo+/-`
${c1}        `sMMMMMMMm:      :dMMMMMMMs`
       `hMMMMMMd/          /dMMMMMMh`
         `://:`              `://:`
```

**Comment by GamerSource** [comment]
> Does Neofetch not detect Proxmox correctly without the extra detection stuff?

No, this is needed. Proxmox VE ships Debians original base-files which provides Debians upstream /etc/os_release.

> You have too many color variables, the art can be simplified.
>   Color formatting follows line breaks.

I saw that but kept it in sync with the screenfetch one, was less work.
But, yes yours is better, I'll adapt this and fix the line length.

Thanks for the feedback!

**Comment by GamerSource** [comment]
Hi @dylanaraps , just FYI I force pushed changes to address you comments.
I.e. keep lines shorter and use your ASCII art adaption which switches only colors when really necessary.
There was a small visual bug in it (one color missed) - I'm noting this just so if one stumbles over this and wants to use it for whichever reason he is aware of this.
Nonetheless much thanks for your feedback, could you take another look to see if now everything is up to neofetchs standards, if you get some spare time?

**Comment by dylanaraps** [comment]
I’ll take a look when I get home from uni. Thanks for making the changes :)

**Comment by GamerSource** [comment]
Hi @dylanaraps  any chance you could look at the current version? WOuld be really great, thanks :)

File: neofetch [hunk 1]
```diff
@@ -84,6 +84,12 @@ get_distro() {
                     *) distro="Siduction ($(lsb_release -sic))"
                 esac
 
+            elif [[ -x "/usr/bin/pveversion" ]]; then
```
Comment (dylanaraps) [inline neofetch]: Change this to `elif type -p pveversion >/dev/null; then`. It's safer to not hardcode the path.

File: neofetch [hunk 2]
```diff
@@ -84,6 +84,12 @@ get_distro() {
...
+                case "$distro_shorthand" in
+                    "on" | "tiny") distro="Proxmox VE" ;;
+                    *) distro="Proxmox VE $(pveversion | grep -oP 'pve-manager\/\K\d+\.\d+')"
```
Comment (dylanaraps) [inline neofetch]: What's the output of `pveversion` on its own? Grep may not be needed if we can use bash builtins.

File: neofetch (hunk 1 above)
Comment (wget) [reply neofetch]: :+1:

File: neofetch [hunk 3]
```diff
@@ -3534,6 +3540,11 @@ get_distro_colors() {
             ascii_file="porteus"
         ;;
 
+        "Proxmox VE"*)
+            set_colors 7 202
```
Comment (wget) [inline neofetch]: Are we authorized to use 24 bits based colors here? The color chosen here is not really the same orange color as Proxmox.
https://en.wikipedia.org/wiki/ANSI_escape_code#24-bit
If not, maybe we could add a note to the README or the wiki in order to help newscomers?

**Comment by wget** [comment]
I made a new pull request including the aforementioned requested changes: https://github.com/dylanaraps/neofetch/pull/1054

The null pull request has been rebased on the latest master.

cc @GamerSource @dylanaraps

**Pull request closed without merging**